src/AspNetCoreModuleV2/InProcessRequestHandler/inprocessapplication.cpp

```diff
@@ -69,6 +69,16 @@ IN_PROCESS_APPLICATION::Stop(bool fServerInitiated)
     HRESULT hr = S_OK;
     CHandle  hThread;
     DWORD    dwThreadStatus = 0;
+
+    SRWExclusiveLock stopLock(m_stateLock);
+
+    if (m_fStopCalled)
```

Comment:
It seems like some applications exit early from `Stop()` if `m_fStopCalled` is set, but not all.  Why is this needed now at all?  Why wasn't it needed before this change?

Reply:
Applications that actually do something in stop (stop managed/kill the process) have to prevent double entrance because we call stop both from OnAppOflline and from ApplicationInfo now.

We call Stop from ApplicationInfo to make sure application stops before we start serving app_offline and files are unlocked.